StarRocksFE start failed

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
 StarRocksFE start failed 

### Expected behavior (Required)
com.starrocks.common.DdlException: failed to load AuthorizationManager!
	at com.starrocks.privilege.AuthorizationMgr.load(AuthorizationMgr.java:1804) ~[starrocks-fe.jar:?]
	at com.starrocks.server.GlobalStateMgr.loadRBACPrivilege(GlobalStateMgr.java:1745) ~[starrocks-fe.jar:?]
	at com.starrocks.server.GlobalStateMgr.loadImage(GlobalStateMgr.java:1596) ~[starrocks-fe.jar:?]
	at com.starrocks.server.GlobalStateMgr.initialize(GlobalStateMgr.java:1082) ~[starrocks-fe.jar:?]
	at com.starrocks.StarRocksFE.start(StarRocksFE.java:133) ~[starrocks-fe.jar:?]
	at com.starrocks.StarRocksFE.main(StarRocksFE.java:84) ~[starrocks-fe.jar:?]
Caused by: com.starrocks.privilege.PrivilegeException: role root is not mutable!
	at com.starrocks.privilege.RolePrivilegeCollectionV2.assertMutable(RolePrivilegeCollectionV2.java:74) ~[starrocks-fe.jar:?]
	at com.starrocks.privilege.RolePrivilegeCollectionV2.grant(RolePrivilegeCollectionV2.java:135) ~[starrocks-fe.jar:?]
	at com.starrocks.privilege.DefaultAuthorizationProvider.upgradePrivilegeCollection(DefaultAuthorizationProvider.java:244) ~[starrocks-fe.jar:?]
	at com.starrocks.privilege.AuthorizationMgr.load(AuthorizationMgr.java:1788) ~[starrocks-fe.jar:?]
	... 5 more
### Real behavior (Required)

### StarRocks version (Required)
 SR3.1

--- a/fe/fe-core/src/main/java/com/starrocks/privilege/AuthorizationMgr.java
+++ b/fe/fe-core/src/main/java/com/starrocks/privilege/AuthorizationMgr.java
@@ -218,8 +218,8 @@ public class AuthorizationMgr {
 
     // called by initBuiltinRolesAndUsers()
     protected void initPrivilegeCollections(PrivilegeCollectionV2 collection, ObjectType objectType,
-                                          List<PrivilegeType> actionList,
-                                          List<String> tokens, boolean isGrant) throws PrivilegeException {
+                                            List<PrivilegeType> actionList,
+                                            List<String> tokens, boolean isGrant) throws PrivilegeException {
         List<PEntryObject> object;
         if (tokens != null) {
             object = Collections.singletonList(provider.generateObject(objectType, tokens, globalStateMgr));
@@ -1729,7 +1729,9 @@ public class AuthorizationMgr {
                     }
 
                     // upgrade meta to current version
-                    ret.provider.upgradePrivilegeCollection(collection, ret.pluginId, ret.pluginVersion);
+                    if (!userIdentity.equals(UserIdentity.ROOT)) {
+                        ret.provider.upgradePrivilegeCollection(collection, ret.pluginId, ret.pluginVersion);
+                    }
                     ret.userToPrivilegeCollection.put(userIdentity, collection);
                 }
                 // 1 json for num roles
@@ -1777,7 +1779,6 @@ public class AuthorizationMgr {
                         rolePrivilegeCollection.typeToPrivilegeEntryList
                                 = builtInRolePrivilegeCollection.typeToPrivilegeEntryList;
                     } else {
-
                         Map<ObjectTypeDeprecate, List<PrivilegeEntry>> m = collectionDeprecate.getTypeToPrivilegeEntryList();
                         for (Map.Entry<ObjectTypeDeprecate, List<PrivilegeEntry>> e : m.entrySet()) {
                             rolePrivilegeCollection.getTypeToPrivilegeEntryList().put(e.getKey().toObjectType(), e.getValue());
@@ -1785,7 +1786,9 @@ public class AuthorizationMgr {
                     }
 
                     // upgrade meta to current version
-                    ret.provider.upgradePrivilegeCollection(rolePrivilegeCollection, ret.pluginId, ret.pluginVersion);
+                    if (!PrivilegeBuiltinConstants.IMMUTABLE_BUILT_IN_ROLE_IDS.contains(roleId)) {
+                        ret.provider.upgradePrivilegeCollection(rolePrivilegeCollection, ret.pluginId, ret.pluginVersion);
+                    }
                     ret.roleIdToPrivilegeCollection.put(roleId, rolePrivilegeCollection);
                 }
             } catch (SRMetaBlockEOFException eofException) {

--- a/fe/fe-core/src/test/java/com/starrocks/privilege/AuthorizationMgrTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/privilege/AuthorizationMgrTest.java
@@ -1752,45 +1752,51 @@ public class AuthorizationMgrTest {
         setCurrentUserAndRoles(ctx, UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv", "%"));
         new NativeAccessControl().checkCatalogAction(ctx.getCurrentUserIdentity(), ctx.getCurrentRoleIds(),
                 "hive_catalog_1", PrivilegeType.CREATE_DATABASE);
-
-        AuthorizationMgr authorizationMgr = GlobalStateMgr.getCurrentState().getAuthorizationMgr();
-        UserPrivilegeCollectionV2 userPrivilegeCollectionV2 = authorizationMgr.getUserPrivilegeCollectionUnlockedAllowNull(
-                UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv", "%"));
-
         Assert.assertThrows(AccessDeniedException.class, () ->
                 new NativeAccessControl().checkCatalogAction(ctx.getCurrentUserIdentity(), ctx.getCurrentRoleIds(),
                         "hive_catalog_1", PrivilegeType.USAGE));
-        authorizationMgr.provider.upgradePrivilegeCollection(userPrivilegeCollectionV2, (short) 1, (short) 1);
-        authorizationMgr.checkAction(userPrivilegeCollectionV2, ObjectType.CATALOG, PrivilegeType.USAGE,
-                Lists.newArrayList("hive_catalog_1"));
 
         DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(
                 "create user test_upgrade_priv_2", ctx), ctx);
         DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(
                 "grant create table on database db to test_upgrade_priv_2", ctx), ctx);
         setCurrentUserAndRoles(ctx, UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv_2", "%"));
-        userPrivilegeCollectionV2 = authorizationMgr.getUserPrivilegeCollectionUnlockedAllowNull(
-                UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv_2", "%"));
         Assert.assertThrows(AccessDeniedException.class, () ->
                 new NativeAccessControl().checkCatalogAction(ctx.getCurrentUserIdentity(), ctx.getCurrentRoleIds(),
                         "hive_catalog_1", PrivilegeType.USAGE));
-        authorizationMgr.provider.upgradePrivilegeCollection(userPrivilegeCollectionV2, (short) 1, (short) 1);
-        authorizationMgr.checkAction(userPrivilegeCollectionV2, ObjectType.CATALOG, PrivilegeType.USAGE,
-                Lists.newArrayList("hive_catalog_1"));
 
         DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(
                 "create user test_upgrade_priv_3", ctx), ctx);
         DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(
                 "grant select on table db.tbl to test_upgrade_priv_3", ctx), ctx);
         setCurrentUserAndRoles(ctx, UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv_3", "%"));
-        userPrivilegeCollectionV2 = authorizationMgr.getUserPrivilegeCollectionUnlockedAllowNull(
-                UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv_3", "%"));
-
         Assert.assertThrows(AccessDeniedException.class, () ->
                 new NativeAccessControl().checkCatalogAction(ctx.getCurrentUserIdentity(), ctx.getCurrentRoleIds(),
                         "hive_catalog_1", PrivilegeType.USAGE));
-        authorizationMgr.provider.upgradePrivilegeCollection(userPrivilegeCollectionV2, (short) 1, (short) 1);
-        authorizationMgr.checkAction(userPrivilegeCollectionV2, ObjectType.CATALOG, PrivilegeType.USAGE,
+
+
+        GlobalStateMgr masterGlobalStateMgr = ctx.getGlobalStateMgr();
+        UtFrameUtils.PseudoJournalReplayer.resetFollowerJournalQueue();
+        UtFrameUtils.PseudoImage image = new UtFrameUtils.PseudoImage();
+        masterGlobalStateMgr.saveRBACPrivilege(image.getDataOutputStream());
+        masterGlobalStateMgr.loadRBACPrivilege(image.getDataInputStream());
+
+        AuthorizationMgr authorizationMgr = GlobalStateMgr.getCurrentState().getAuthorizationMgr();
+        UserPrivilegeCollectionV2 up1 = authorizationMgr.getUserPrivilegeCollectionUnlockedAllowNull(
+                UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv", "%"));
+        UserPrivilegeCollectionV2 up2 = authorizationMgr.getUserPrivilegeCollectionUnlockedAllowNull(
+                UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv_2", "%"));
+        UserPrivilegeCollectionV2 up3 = authorizationMgr.getUserPrivilegeCollectionUnlockedAllowNull(
+                UserIdentity.createAnalyzedUserIdentWithIp("test_upgrade_priv_3", "%"));
+
+        authorizationMgr.checkAction(up1, ObjectType.CATALOG, PrivilegeType.USAGE,
                 Lists.newArrayList("hive_catalog_1"));
+        authorizationMgr.checkAction(up2, ObjectType.CATALOG, PrivilegeType.USAGE,
+                Lists.newArrayList("hive_catalog_1"));
+        authorizationMgr.checkAction(up3, ObjectType.CATALOG, PrivilegeType.USAGE,
+                Lists.newArrayList("hive_catalog_1"));
+
+        masterGlobalStateMgr.saveRBACPrivilege(image.getDataOutputStream());
+        masterGlobalStateMgr.loadRBACPrivilege(image.getDataInputStream());
     }
 }

